app/javascript/components/concept-map/Concept.tsx

```diff
@@ -60,44 +70,44 @@ export const Concept = ({
     classes.push('not-started')
   }
 
-  if (slug === 'basics') {
-    console.log({
-      slug,
-      name,
-      web_url,
-      handleEnter,
-      handleLeave,
-      status,
-      isActive,
-      exercises,
-      exercisesCompleted,
-    })
-  }
-
   return (
-    <a
-      ref={conceptRef}
-      href={web_url}
-      id={conceptSlugToId(slug)}
-      className={classes.join(' ')}
-      data-concept-slug={slug}
-      data-concept-status={status}
-      onMouseEnter={wrapAnimationFrame(handleEnter)}
-      onMouseLeave={wrapAnimationFrame(handleLeave)}
-    >
-      <div className="display">
-        <div className="name">{name}</div>
-        <CompleteIcon show={hasExercises && exercises === exercisesCompleted} />
-      </div>
-      <PureExerciseProgressBar
-        completed={exercisesCompleted}
-        exercises={exercises}
-        hidden={!hasExercises || !isStarted}
+    <div onFocus={() => setHasFocus(true)} onBlur={() => setHasFocus(false)}>
+      <a
+        ref={conceptRef}
+        href={web_url}
+        id={conceptSlugToId(slug)}
+        className={classes.join(' ')}
+        data-concept-slug={slug}
+        data-concept-status={status}
+        onMouseEnter={wrapAnimationFrame(handleEnter)}
+        onMouseLeave={wrapAnimationFrame(handleLeave)}
+      >
+        <div className="display">
+          <div className="name">{name}</div>
+          <CompleteIcon
+            show={hasExercises && exercises === exercisesCompleted}
+          />
+        </div>
+        <PureExerciseProgressBar
+          completed={exercisesCompleted}
+          exercises={exercises}
+          hidden={!hasExercises || !isStarted}
+        />
+      </a>
+      <Tooltips.Concept
+        endpoint={tooltip_url}
+        parent={conceptRef.current}
+        requestToShow={isActiveHover || hasFocus}
       />
-    </a>
+    </div>
   )
 }
 
+// focus on concept / focus-blur
+// focus on tooltip / focus-blur
+// focus inside tooltip
+// no focus
```

Comment:
Can these be removed now?

Reply:
Ya, I'll clean them up when this is ready for final review. It's only half working so far

Reply:
OK :)